src/main/java/pl/simpleascoding/tutoringplatform/service/teacher/TeacherServiceImpl.java

```diff
@@ -0,0 +1,64 @@
+package pl.simpleascoding.tutoringplatform.service.teacher;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import pl.simpleascoding.tutoringplatform.domain.user.Role;
+import pl.simpleascoding.tutoringplatform.domain.user.RoleType;
+import pl.simpleascoding.tutoringplatform.domain.user.User;
+import pl.simpleascoding.tutoringplatform.dto.SignAsTeacherDTO;
+import pl.simpleascoding.tutoringplatform.dto.UserDTO;
+import pl.simpleascoding.tutoringplatform.exception.UserIsAlreadyATeacherException;
+import pl.simpleascoding.tutoringplatform.repository.RoleRepository;
+import pl.simpleascoding.tutoringplatform.service.user.UserModelMapper;
+import pl.simpleascoding.tutoringplatform.service.user.UserService;
+
+import javax.transaction.Transactional;
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class TeacherServiceImpl implements TeacherService {
+
+    private final RoleRepository roleRepository;
+    private final UserService userService;
+    private final UserModelMapper userModelMapper;
+
+    @Override
+    @Transactional
+    public void addTeacherRoleToUser(SignAsTeacherDTO requestDTO) {
+        User user = userService.getUserByUsername(requestDTO.username());
+        Role roleTeacher = createTeacherRoleEntity();
+        isUserAlreadyTeacher(user, roleTeacher);
```

Comment:
Name of this method is wrong. Name of the method right now is the question - that mean it should return some value. But it doesn't.